### PR TITLE
Make switches/0 and aliases/0 optional command callbacks

### LINKED ISSUE #120: Make `switches/0` an optional callback
After https://github.com/rabbitmq/rabbitmq-cli/commit/59ba4d91b379fb76e829f7d037f6c9a04146e34f it becomes confusing that some commands won't have to define any switches since they only support `-p [vhost]` (which is auto-defined by the tool).

[delete-shovel](https://github.com/rabbitmq/rabbitmq-shovel/blob/master/src/Elixir.RabbitMQ.CLI.Ctl.Commands.DeleteShovelCommand.erl#L41) is a good example of this.

The `switches/0` callback therefore should probably be made optional.

### PR DESCRIPTION
**Pull request by michaelklishin**
Fixes #120. Same as #124 but doesn't include any unrelated changes
and thus suffer from #126.

### CODE REVIEW

File: lib/rabbitmq/cli/command_behaviour.ex [hunk 1]
```diff
@@ -20,16 +20,20 @@ defmodule RabbitMQ.CLI.CommandBehaviour do
...
-  @callback aliases() :: Keyword.t
   # Coerces run/2 return value into the standard command output form
   # that is then formatted, printed and returned as an exit code.
   # There is a default implementation for this callback in DefaultOutput module
   @callback output(any, Map.t) :: :ok | {:ok, any} | {:stream, Enum.t} |
                                   {:error, ExitCodes.exit_code, [String.t]}
   @optional_callbacks formatter: 0,
                       scopes: 0,
-                      usage_additional: 0
+                      usage_additional: 0,
```
Comment (acogoluegnes) [inline lib/rabbitmq/cli/command_behaviour.ex]: Is there a way to provide a default implementation for an optional callback? The [documentation](http://elixir-lang.org/docs/stable/elixir/behaviours) mentions it, but I couldn't find a proper example and couldn't come up with something that would work.

Comment (michaelklishin) [inline lib/rabbitmq/cli/command_behaviour.ex]: There is no inheritance in Elixir and this is not a problem with polymorphism, so protocols do not fit either.

Comment (michaelklishin) [inline lib/rabbitmq/cli/command_behaviour.ex]: [This blog post about Elixir 1.3](http://tuvistavie.com/2016/elixir-1-3/#support-of-otp-optionalcallback) mentions default implementations but I cannot find any specific examples.

Comment (michaelklishin) [inline lib/rabbitmq/cli/command_behaviour.ex]: I guess the point of "default implementation" means that a behaviour could also define a function to fall back to but that wouldn't help us avoid `function_exported?/4` entirely.

Comment (acogoluegnes) [inline lib/rabbitmq/cli/command_behaviour.ex]: OK. I let @hairyhum have a look at the PR before merging.

Comment (hairyhum) [inline lib/rabbitmq/cli/command_behaviour.ex]: There is something like "default implementation" by using [Elixir mixins](http://elixir-lang.org/getting-started/alias-require-and-import.html#use)
But that would work for Elixir commands only, and Erlang commands will have to define all the callbacks.

**Comment by michaelklishin** [comment]
Anything specific left to address in this PR?

**Comment by hairyhum** [comment]
That should be all

**Comment by michaelklishin** [comment]
Thanks @hairyhum.

**Pull request merged**